cpu-ffmpeg-performance-benchmarks.md

```diff
@@ -0,0 +1,145 @@
+
+
+### Machine shapes
+
+For performance and price comparisons, we test on VMs with 8 vCPUs and 16 GB of RAM. For most machine families, this is a predefined shape of `xx-highcpu-8`. For others, we need to define a custom configuration.
+
+## Resolutions
+
+For the purposes of this test, we work with the following resolutions:
+
+| Source | Resolution | Other names |
+| ------ | ---------- | ----------- |
+Raw (source video) | 3840x2160 | 2160p, 4k
+SD | 854x480 | 480p, Standard Definition
+HD | 1920x1080 | 1080p, High Definition
+
+## Video source asset
+
+We use the open source animation project [Big Buck Bunny](https://peach.blender.org/) (sponsored by Blender). This project is widely used as test footage in many content creation industries. The highest resolution available is 4k. The duration of the video is ~12 minutes. The original download can be found [here](https://download.blender.org/demo/movies/BBB/) (2160p, 30fps, mp4).
+
+To test a long-format transcode, we generated a 160-minute version of the original by looping the video multiple times, written to a single file. We then performed transcoding benchmarks on this video to both HD and SD resolutions.
+
+## Testing methodology
+
+We're using the same testing framework built for the blog post [Benchmarking Rendering Software](https://cloud.google.com/blog/topics/developers-practitioners/benchmarking-rendering-software-compute-engine) (GitHub repo [here](https://github.com/adrian-a-graham/benchmarking-rendering-software)). This framework performs the following:
+
+1. Run bash script from a local terminal to generate VMs based on a config file.
+
+    ```
+    bin/run_benchmark_ffmpeg.sh \
+      --config [CONFIG_FILE] \
+      --benchmark [BENCHMARK] \
+      --resolution [RESOLUTION] \
+      [--verbose|--noop]
+    ```
+    This will create one VM per line in `[CONFIG_FILE]` with a startup script for the specified `[BENCHMARK]`.
+
+2. The startup script downloads the benchmark script. This script is self-launching, and draws variables from the VM metadata:
+
+    | Key | Value |
+    | --- | ----- |
+    `startup-script-url` | `gs://path/to/script/[BENCHMARK]-[OS]-startup.sh`
+    session | `[BENCHMARK]-[DATE]-[RESOLUTION]`
+    delete | Boolean, whether to delete the VM on benchmark completion. Default=True
+    resolution | 'sd' or 'hd'
+
+3. The benchmark script runs and performs the following:
+    1. Install `ffmpeg` and supporting libraries.
+    1. Download source video file to transcode.
+    1. Run `ffmpeg` command, looping 4 times, recording each duration.
+        1. The first iteration is discarded to ensure disk caching performance isn't factored into the overall calculation.
+    1. An average duration is calculated from the remaining 5 runs.
+    1. Metrics are written locally and pushed to a Google Cloud Storage bucket in JSON format.
+
+## ffmpeg command
+
+We transcoded the source video using the ffmpeg configuration:
+
+    ffmpeg \
```

Comment:
Can you put this multi-line command in a code block? ie: use triple back-ticks before and after the block